tests/unit/s2n_client_hello_recv_test.c

```diff
@@ -97,24 +103,53 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_stuffer_write(&server_conn->handshake.io, &client_hello));
         EXPECT_SUCCESS(s2n_client_hello_recv(server_conn));
 
-        EXPECT_EQUAL(server_conn->server_protocol_version, S2N_TLS12);
+        EXPECT_EQUAL(server_conn->server_protocol_version, i == 0 ? S2N_TLS12 : S2N_TLS13);
         EXPECT_EQUAL(server_conn->actual_protocol_version, S2N_TLS12);
         EXPECT_EQUAL(server_conn->client_protocol_version, S2N_TLS12);
         EXPECT_EQUAL(server_conn->client_hello_version, S2N_SSLv2);
         EXPECT_EQUAL(server_conn->client_hello.parsed, 1);
 
         s2n_connection_free(server_conn);
+
+        EXPECT_SUCCESS(s2n_disable_tls13());
     }
-    
-    /* Test that a tls12 client legacy version and tls12 server version 
+
+    /* Test that a tls12 client legacy version and tls12 server version
```

Comment:
These tests check in a very concrete fashion whether TLS1.3 is working/not working when TLS1.2 is active/not active.
You might be able to generalize those in the future, which might allow to remove some code make things more extensible. However it's not super important.

For every protocol version
- Try against a client which has every other protocol version enabled, which should yield the other protocol.
- If the desired protocol is enabled, you should get that one.